CI failing on mac

The mac CI is failing for it appears both cmdstan and rstan models. Using a mac locally I see no issues and there are no reported mac issues either here or upstream in rstan or brms suggesting a CI issue.

This is the error - possibly related to boost/BH?

```c
Compilation ERROR, function(s)/method(s) not created!
Error in compileCode(f, code, language = language, verbose = verbose) : 
  cialization 'rstan::(anonymous namespace)::command<model96539299c7b__namespace::model96539299c7b_, boost::random::additive_combine_engine<boost::random::linear_congruential_engine<unsigned int, 40014, 0, 2147483563>, boost::random::linear_congruential_engine<unsigned int, 40692, 0, 2147483399>>>' requested here
 1215 |     ret = command(args, model_, holder, names_oi_tidx_,
      |           ^
file9655ae06c8e.cpp:2392:68: note: in instantiation of member function 'rstan::stan_fit<model96539299c7b__namespace::model96539299c7b_, boost::random::additive_combine_engine<boost::random::linear_congruential_engine<unsigned int, 40014, 0, 2147483563>, boost::random::linear_congruential_engine<unsigned int, 40692, 0, 2147483399>>>::call_sampler' requested here
 2392 |           &rstan::stan_fit<stan_model, boost::random::ecuyer1988>::call_sampler)
      |                                                                    ^
14 warnings and 4 errors generated.
```

The latent model gives

```c
Error in compileCode(f, code, language = language, verbose = verbose) : 
  signed int, 40014, 0, 2147483563>, linear_congruential_engine<unsigned int, 40692, 0, 2147483399>> &&' for 1st argument
   50 | class additive_combine_engine
      |       ^~~~~~~~~~~~~~~~~~~~~~~
/Users/eidesfun/Library/R/arm64/4.4/library/BH/include/boost/random/additive_combine.hpp:79:48: note: explicit constructor is not a candidate
   79 |     BOOST_RANDOM_DETAIL_ARITHMETIC_CONSTRUCTOR(additive_combine_engine,
      |                                                ^
/Users/eidesfun/Library/R/arm64/4.4/library/BH/include/boost/random/additive_combine.hpp:97:46: note: explicit constructor is not a candidate
   97 |     BOOST_RANDOM_DETAIL_SEED_SEQ_CONSTRUCTOR(additive_combine_engine,
      |                                              ^
file9651b34186e.cpp:521:11: warning: variable 'pos__' set but not used [-Wunused-but-set-variable]
  521 |       int pos__ = std::numeric_limits<int>::min();
      |           ^
12 warnings and 1 error generated.
```

No, and actually open looking more closely epidist with cmdstanr doesn't work either with the marginal model. Same error:

```c
. + Compiling Stan program...
/var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpeMkxGX/model-8dc17775757.hpp:736:47: error: non-constant-expression cannot be narrowed from type 'int64_t' (aka 'long long') to 'int' in initializer list [-Wc++11-narrowing]
  736 |         std::vector<int>{dist_id, primary_id, stan:
:math::size(params),
      |                                               ^~~~~~~~~~~~~~~~~~~~~~~~
/var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpeMkxGX/model-8dc17775757.hpp:822:18: note: in instantiation of function template specialization 'primarycensored_lognormal_model_namespace::primarycensored_cdf<int, std::vector<double>, int, std::vector<double>, nullptr>' requested here
  822 |                  primarycensored_cdf(d, dist_id, params, pwindow,
      |                  ^
/var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpeMkxGX/model-8dc17775757.hpp:906:21: note: in instantiation of function template specialization 'primarycensored_lognormal_model_namespace::primarycensored_lcdf<int, std::vector<double>, int, std::vector<double>, nullptr>' requested here
  906 |     log_cdf_upper = primarycensored_lcdf(d_upper, dist_id, params, pwindow,
      |                     ^
/var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpeMkxGX/model-8dc17775757.hpp:2321:13: note: in instantiation of function template specialization 'primarycensored_lognormal_model_namespace::primarycensored_lpmf<false, int, std::vector<double>, int, std::vector<double>, nullptr>' requested here
 2321 |             primarycensored_lpmf<false>(
      |             ^
/var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpeMkxGX/model-8dc17775757.hpp:2593:12: note: in instantiation of function template specialization 'primarycensored_lognormal_model_namespace::primarycensored_lognormal_model::log_prob_impl<false, false, Eigen::Matrix<double, -1, 1>, Eigen::Matrix<int, -1, 1>, nullptr, nullptr, nullptr>' requested here
 2593 |     return log_prob_impl<propto__, jacobian__>(params_r, params_i, pstream);
      |            ^
/Users/eidesfun/.cmdstan/cmdstan-2.36.0/stan/src/stan/model/model_base_crtp.hpp:93:50: note: in instantiation of function template specialization 'primarycensored_lognormal_model_namespace::primarycensored_lognormal_model::log_prob<false, false, double>' requested here
   
93 |     return static_cast<const M*>(this)->template log_prob<false, false, double>(
      |                                                  ^
/var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpeMkxGX/model-8dc17775757.hpp:2188:3: note: in instantiation of member function 'stan::model::model_base_crtp<primarycensored_lognormal_model_namespace::primarycensored_lognormal_model>::log_prob' requested here
 2188 |   ~primarycensored_lognormal_model() {}
      |   ^

/var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpeMkxGX/model-8dc17775757.hpp:736:47: note: insert an explicit cast to silence this issue
  736 |         std::vector<int>{dist_id, primary_id, stan::math::size(params),
      |                                               ^~~~~~~~~~~~~~~~~~~~~~~~
      |                                               static_cast<int>(       )
/var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpeMkxGX/model-8dc17775757.hpp:737:11: error: non-constant-expression cannot be narrowed from type 'int64_t' (aka 'long long') to 'int' in initializer list [-Wc++11-narrowing]
  737 |           stan::math::size(primary_params)}, "assigning variable ids");
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpeMkxGX/model-8dc17775757.hpp:737:11: note: insert an explicit cast to silence this issue
  737 |           stan::math::size(primary_params)}, "assigning variable ids");
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |           static_cast<int>(               )

/var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpeMkxGX/model-8dc17775757.hpp:736:47: error: non-constant-expression cannot be narrowed from type 'int64_t' (aka 'long long') to 'int' in initializer list [-Wc++11-narrowing]
  736 |         std::vector<int>{dist_id, primary_id, stan::math::size(params),
      |                                               ^~~~~~~~~~~~~~~~~~~~~~~~
/var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpeMkxGX/model-8dc17775757.hpp:822:18: note: in instantiation of function template specialization 'primarycensored_lognormal_model_namespace::primarycensored_cdf<int, std::vector<stan::math::var_value<double>>, int, std::vector<double>, nullptr>' requested here
  822 |                  primarycensored_cdf(d, dist_id, params, pwindow,
      |                  ^
/var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpeMkxGX/model-8dc17775757.hpp:906:21: note: in instantiation of function template specialization 'primarycensored_lognormal_model_namespace::primarycensored_lcdf<int, std::vector<stan::math::var_value<double>>, int, std::vector<double>, nullptr>' requested here
  906 |     log_cdf_upper = primarycensored_lcdf(d_upper, dist_id, params, pwindow,
      |                     ^
/var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpeMkxGX/model-8dc17775757.hpp:2381:13: note: in instantiation of function template specialization 'primarycensored_lognormal_model_namespace::primarycensored_lpmf<false, int, std::vector<stan::math::var_value<double>>, int, std::vector<double>, nullptr>' requested here
 2381 |             primarycensored_lpmf<false>(
      |             ^
/var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpeMkxGX/model-8dc17775757.hpp:2593:12: note: in instantiation of function template specialization 'primarycensored_lognormal_model_namespace::primarycensored_lognormal_model::log_prob_impl<false, false, Eigen::Matrix<stan::math::var_value<double>, -1, 1>, Eigen::Matrix<int, -1, 1>, nullptr, nullptr, nullptr>' requested here
 
2593 |     return log_prob_impl<propto__, jacobian__>(params_r, params_i, pstream);
      |            ^

/Users/eidesfun/.cmdstan/cmdstan-2.36.0/stan/src/stan/model/model_base_crtp.hpp:98:50: note: in instantiation of function template specialization 'primarycensored_lognormal_model_namespace::primarycensored_lognormal_model::log_prob<false, false, stan::math::var_value<double>>' requested here
   98 |     return static_cast<const M*>(this)->template log_prob<false, false>(theta,
      |                                                  ^
/var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpeMkxGX/model-8dc17775757.hpp:2188:3: note: in instantiation of member function 'stan::model::model_base_crtp<primarycensored_lognormal_model_namespace::primarycensored_lognormal_model>::log_prob' requested here
 2188 |   ~primarycensored_lognormal_model() {}
      |   ^
/var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpeMkxGX/model-8dc17775757.hpp:736:47: note: insert an explicit cast to silence this issue
  736 |         std::vector<int>{dist_id, primary_id, stan::math::size(params),
      |                                               ^~~~~~~~~~~~~~~~~~~~~~~~
      |                                               static_cast<int>(       )
/var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpeMkxGX/model-8dc17775757.hpp:737:11: error: non-constant-expression cannot be narrowed from type 'int64_t' (aka 'long long') to 'int' in initializer list [-Wc++11-narrowing]
  737 |           stan::math::size(primary_params)}, "assigning variable ids");
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpeMkxGX/model-8dc17775757.hpp:737:11: note: insert an explicit cast to silence this issue
  737 |           stan::math::size(primary_params)}, "assigning variable ids");
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |           static_cast<int>(               )

4 errors generated.
```

--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -71,7 +71,7 @@ Suggests:
     RcppEigen (>= 0.3.3.3.0)
 Remotes:
     stan-dev/cmdstanr,
-    epinowcast/primarycensored@get-rid-of-size
+    epinowcast/primarycensored
 Config/Needs/website:
     r-lib/pkgdown,
     epinowcast/enwtheme

--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -70,7 +70,8 @@ Suggests:
     Rcpp (>= 0.12.0),
     RcppEigen (>= 0.3.3.3.0)
 Remotes:
-    stan-dev/cmdstanr
+    stan-dev/cmdstanr,
+    epinowcast/primarycensored@get-rid-of-size
 Config/Needs/website:
     r-lib/pkgdown,
     epinowcast/enwtheme

--- a/NEWS.md
+++ b/NEWS.md
@@ -21,12 +21,12 @@ Development version of `epidist`.
 ## Documentation
 
 - Brings the README into line with `epinowcast` standards. See #467.
-- Switched over to using the marginal model as default in documentation. See #426.
-- Added helper functions for new variables to avoid code duplication in vignettes. See #426.
+- Switched over to using the marginal model as default in the documentation. See #426.
+- Added a helper functions for new variables to avoid code duplication in vignettes. See #426.
 
 ## Bugs
 
-- Switch to using a patched of `primarycensored` that doesn't make use of `size()`. This fixes some Mac compilation edge cases. See #524.
+- Switched to using a patched of `primarycensored` that doesn't make use of `size()`. This fixes some Mac compilation edge cases. See #524.
 
 # epidist 0.1.0
 

--- a/NEWS.md
+++ b/NEWS.md
@@ -24,6 +24,10 @@ Development version of `epidist`.
 - Switched over to using the marginal model as default in documentation. See #426.
 - Added helper functions for new variables to avoid code duplication in vignettes. See #426.
 
+## Bugs
+
+- Switch to using a patched of `primarycensored` that doesn't make use of `size()`. This fixes some Mac compilation edge cases. See #524.
+
 # epidist 0.1.0
 
 This is the first minor release of `epidist` intended for early test users of the package.

